tool/tsh/tsh.go

```diff
@@ -939,6 +955,61 @@ func onBenchmark(cf *CLIConf) {
 		fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
 		os.Exit(255)
 	}
+	if cf.BenchExport {
+		var fullPath string
+		timeStamp := fmt.Sprint(time.Now().Format("2006-01-02_15:04:05"))
+		suffix := fmt.Sprint("/latency_profile_" + timeStamp + ".txt")
+		// If path is specified
+		if cf.BenchExportPath != "" {
+			usr, err := user.Current()
+			if err != nil {
+				fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
+			}
+			dir := usr.HomeDir
+			// Expand tilde if used
+			if cf.BenchExportPath == "~" {
+				cf.BenchExportPath = dir
+			} else if strings.HasPrefix(cf.BenchExportPath, "~/") {
+				cf.BenchExportPath = filepath.Join(dir, cf.BenchExportPath[2:])
+			}
+			// Check if path is valid
+			if _, err := os.Stat(cf.BenchExportPath); err != nil {
+				if os.IsNotExist(err) {
+					fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
+				}
```

Comment:
If it doesn't exist, use `os.MkdirAll` to create the directory.
If there's any other kind of error (like permissions), return